docs/concepts/models/python_models.md

```diff
@@ -90,7 +90,40 @@ Optional pre/post-statements allow you to execute SQL commands before and after
 
 For example, pre/post-statements might modify settings or create indexes. However, be careful not to run any statement that could conflict with the execution of another statement if models run concurrently, such as creating a physical table.
 
-Pre- and post-statements are issued with the SQLMesh [`fetchdf` method](../../reference/cli.md#fetchdf) [described above](#execution-context).
+You can set the `pre_statements` and `post_statements` arguments to a list of SQL strings and/or SQLGlot expressions to define the model's pre/post-statements.
+
+``` python linenums="1" hl_lines="6-7"
+def execute(
+    context: ExecutionContext,
+    start: datetime,
+    end: datetime,
+    execution_time: datetime,
+    pre_statements=["SET GLOBAL parameter = 'value';"],
+    post_statements=["@CREATE_INDEX('example.pre_post_statements', id)"],
```

Comment:
This would need to call against the physical table, right? 

Can a user pass `@this_model` instead of model name so it gets automatically resolved to the physical table? 

If not, I think the `@CREATE_INDEX` definition below needs to extract the physical table [from the snapshot](https://sqlmesh.readthedocs.io/en/stable/concepts/macros/sqlmesh_macros/#accessing-snapshots), although I don't know the specific code for that off the top of my head.

Reply:
yes they can and it will be resolved. so I updated the example to use `@this_model` instead to show this